std/experimental/allocator/building_blocks/bitmapped_block.d

```diff
@@ -1746,3 +1760,334 @@ private struct BitVector
+*/
+shared struct SharedBitmappedBlock(size_t theBlockSize,
+    size_t theAlignment = platformAlignment, ParentAllocator = NullAllocator)
+{
+nothrow @nogc:
+    import std.traits : hasMember;
+    import std.typecons : Ternary;
+
+    /**
+    If `blockSize == chooseAtRuntime`, `SharedBitmappedBlock` offers a read/write
+    property `blockSize`. It must be set before any use of the allocator.
+    Otherwise (i.e. `theBlockSize` is a legit constant), `blockSize` is
+    an alias for `theBlockSize`. Whether constant or variable, must also be
+    a multiple of `alignment`. This constraint is `assert`ed statically
+    and dynamically.
+    */
+    static if (theBlockSize != chooseAtRuntime)
+    {
+        alias blockSize = theBlockSize;
+    }
+    else
+    {
+        @property uint blockSize() { return _blockSize; }
+        @property void blockSize(uint s)
+        {
+            assert(_control.length == 0 && s % alignment == 0);
+            _blockSize = s;
+        }
+        private uint _blockSize;
+    }
+
+    static if (is(ParentAllocator == NullAllocator))
+    {
+        private enum parentAlignment = platformAlignment;
+    }
+    else
+    {
+        private alias parentAlignment = ParentAllocator.alignment;
+        static assert(parentAlignment >= ulong.alignof);
+    }
+
+    /**
+    The _alignment offered is user-configurable statically through parameter
+    `theAlignment`, defaulted to `platformAlignment`.
+    */
+    alias alignment = theAlignment;
+
+    // state {
+    /**
+    The _parent allocator. Depending on whether `ParentAllocator` holds state
+    or not, this is a member variable or an alias for
+    `ParentAllocator.instance`.
+    */
+    static if (stateSize!ParentAllocator)
+    {
+        ParentAllocator parent;
+    }
+    else
+    {
+        alias parent = ParentAllocator.instance;
+    }
+
+    private size_t _blocks;
+    private ulong[] _control;
+    private void[] _payload;
+
+    pure nothrow @safe @nogc
+    private size_t totalAllocation(size_t capacity)
+    {
+        auto blocks = capacity.divideRoundUp(blockSize);
+        auto leadingUlongs = blocks.divideRoundUp(64);
+        import std.algorithm.comparison : min;
+        immutable initialAlignment = min(parentAlignment,
+            1U << trailingZeros(leadingUlongs * 8));
+        auto maxSlack = alignment <= initialAlignment
+            ? 0
+            : alignment - initialAlignment;
+        return leadingUlongs * 8 + maxSlack + blockSize * blocks;
+    }
+
+    /**
+    Constructs a block allocator given a hunk of memory, or a desired capacity
+    in bytes.
+
+    $(UL
+    $(LI If `ParentAllocator` is $(REF_ALTTEXT `NullAllocator`, NullAllocator, std,experimental,allocator,building_blocks,null_allocator),
+    only the constructor taking `data` is defined and the user is responsible for freeing `data` if desired.)
+    $(LI Otherwise, both constructors are defined. The `data`-based
```

Comment:
You got four:

![image](https://user-images.githubusercontent.com/4370550/38294418-1247a652-37eb-11e8-9b37-a5c759fc7934.png)